csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,188 @@
+"""
+Inspect Dataset Manager
+-----------------------
+
+Collection of functions handling csv-related operations used in the inspect cli.
+"""
+
+
+import logging
+from pathlib import Path
+from typing import List, Optional
+import pandas as pd
+
+
+from csvcubed.models.inspectsparqlresults import (
+    QubeComponentResult,
+)
+from csvcubed.utils.qb.components import (
+    ComponentField,
+    ComponentPropertyAttributeURI,
+    ComponentPropertyType,
+    get_component_property_as_relative_path,
+)
+from csvcubed.models.inspectdataframeresults import (
+    DatasetObservationsByMeasureUnitInfoResult,
+    DatasetObservationsInfoResult,
+    DatasetSingleMeasureResult,
+)
+from csvcubed.models.csvcubedexception import (
+    CsvToDataFrameLoadFailedException,
+    InvalidNumberOfRecordsException,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _filter_components_from_dsd(
+    components: List[QubeComponentResult],
+    field: ComponentField,
+    filter: str,
+) -> List[QubeComponentResult]:
+    """
+    Filters the components for the given filter.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `List[QubeComponentResult]` - filtered results.
+    """
+
+    return [
+        component
+        for component in components
+        if component.as_dict()[field.value] == filter
+    ]
+
+
+def load_csv_to_dataframe(json_path: Path, csv_path: Path) -> pd.DataFrame:
+    """
+    Loads the csv in given path to a Panda Dataframe.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DataFrame` - Dataframe of the csv.
+    """
+
+    try:
+        dataset_path = json_path.parent / csv_path
+        _logger.debug(f"Dataset path: {dataset_path.absolute()}")
+
+        dataset = pd.DataFrame(pd.read_csv(dataset_path))
+        _logger.info("Successfully loaded csv into dataframe.")
+
+        return dataset
+    except Exception as ex:
+        raise CsvToDataFrameLoadFailedException from ex
+
+
+def get_measure_col_name_from_dsd(
+    components: List[QubeComponentResult],
+) -> Optional[str]:
+    """
+    Identifies the name of measure column.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `Optional[str]`
+    """
+    filtered_components = _filter_components_from_dsd(
+        components,
+        ComponentField.Property,
+        ComponentPropertyAttributeURI.MeasureType.value,
+    )
+
+    return (
+        filtered_components[0].csv_col_title
+        if filtered_components[0] and filtered_components[0].csv_col_title != ""
```

Comment:
Can we fail more gracefully if we find there isn't a measures column?

Ideally we could log an error and return `None`?